cql3/statements/create_keyspace_statement.cc

```diff
@@ -148,34 +148,37 @@ std::optional<sstring> check_restricted_replication_strategy(
     if (!attrs.get_replication_strategy_class()) {
         return std::nullopt;
     }
+
+    static const std::unordered_map<sstring, locator::replication_strategy_type> rs_to_type = {
+            {"org.apache.cassandra.locator.SimpleStrategy", locator::replication_strategy_type::simple},
+            {"org.apache.cassandra.locator.NetworkTopologyStrategy", locator::replication_strategy_type::network_topology},
+            {"org.apache.cassandra.locator.EverywhereStrategy", locator::replication_strategy_type::everywhere_topology},
+            {"org.apache.cassandra.locator.LocalStrategy", locator::replication_strategy_type::local}
+    };
+
     sstring replication_strategy = locator::abstract_replication_strategy::to_qualified_class_name(
         *attrs.get_replication_strategy_class());
-    auto& topology = qp.proxy().get_token_metadata_ptr()->get_topology();
-    // SimpleStrategy is not recommended in any setup which already has - or
-    // may have in the future - multiple racks or DCs. So depending on how
-    // protective we are configured, let's prevent it or allow with a warning:
-    if (replication_strategy == "org.apache.cassandra.locator.SimpleStrategy") {
-        switch(qp.db().get_config().restrict_replication_simplestrategy()) {
-        case db::tri_mode_restriction_t::mode::TRUE:
-            throw exceptions::configuration_exception(
-                "SimpleStrategy replication class is not recommended, and "
-                "forbidden by the current configuration. Please use "
-                "NetworkToplogyStrategy instead. You may also override this "
-                "restriction with the restrict_replication_simplestrategy=false "
-                "configuration option.");
-        case db::tri_mode_restriction_t::mode::WARN:
-            return format("SimpleStrategy replication class is not "
-                "recommended, but was used for keyspace {}. The "
-                "restrict_replication_simplestrategy configuration option "
-                "can be changed to silence this warning or make it into an error.",
-                keyspace);
-        case db::tri_mode_restriction_t::mode::FALSE:
-            // Scylla was configured to allow SimpleStrategy, but let's warn
-            // if it's used on a cluster which *already* has multiple DCs:
-            if (topology.get_datacenter_endpoints().size() > 1) {
-                return "Using SimpleStrategy in a multi-datacenter environment is not recommended.";
-            }
-            break;
```

Comment:
My thumbs-up meant rather: "acknowledge" than "I'll revert it", and yes, it was confusing, sorry.
The reason I removed this check is simply because we've discussed many similar checks related to RF guardrail and didn't decide on any in the end, hence I thought we're going with the minimalistic approach. Also, as you mentioned, we're "generally" warning about SimpleStategy already, because by default it's on the warn list.